File: examples/python/tasks/text_to_speech/text_to_speech.py
from bytez import Bytez
import base64
import os

WORKING_DIR = os.path.dirname(os.path.realpath(__file__))

client = Bytez("YOUR_BYTEZ_KEY_HERE")

model = client.model("suno/bark-small")

model.load()

input_text = "Hello, how are you today?"

result = model.run(input_text)

output_wav = result.get("output_wav")

# Decode the base64 string to bytes
audio_bytes = base64.b64decode(output_wav)

# Write the audio to the local file system
output_path = os.path.join(WORKING_DIR, "output.wav")
with open(output_path, "wb") as audio_file:
    audio_file.write(audio_bytes)

print(f"Audio successfully saved to {output_path}")

File: examples/julia/tasks/text_to_speech/text_to_speech.jl
using Bytez
using Base64
using Printf

WORKING_DIR = dirname(@__FILE__)

client = Bytez.init("YOUR_BYTEZ_KEY_HERE")

model = client.model("suno/bark-small")

model.load()

input_text = "Hello, how are you today?"

result = model.run(input_text)

output_wav = result["output_wav"]

# Decode the base64 string to bytes
audio_bytes = base64decode(output_wav)

# Write the audio to the local file system
output_path = joinpath(WORKING_DIR, "output.wav")
open(output_path, "w") do audio_file
	write(audio_file, audio_bytes)
end

println("Audio successfully saved to $output_path")
